src/core/lib/transport/transport.cc

```diff
@@ -186,6 +186,36 @@ void grpc_transport_stream_op_batch_queue_finish_with_failure(
   GRPC_ERROR_UNREF(error);
 }
 
+void grpc_transport_stream_op_batch_finish_with_failure_without_call_combiner(
```

Comment:
Suggest renaming this to `grpc_transport_stream_op_batch_finish_with_failure_from_transport()`.

Also, please document in the .h file that this should be used only when failing a batch from the transport itself, below the layer of the call combiner or activity code.

Reply:
done